app/controllers/schools/advice/storage_heaters_controller.rb

```diff
@@ -1,17 +1,78 @@
 module Schools
   module Advice
     class StorageHeatersController < AdviceBaseController
+      before_action :set_seasonal_analysis, only: [:insights, :analysis]
+      before_action :set_annual_usage_breakdown, only: [:insights, :analysis]
+      before_action :set_usage_categories, only: [:analysis]
+      before_action :set_heating_thermostatic_analysis, only: [:insights, :analysis]
+
       def insights
       end
 
       def analysis
+        @analysis_dates = analysis_dates
       end
 
       private
 
+      def set_heating_thermostatic_analysis
+        @heating_thermostatic_analysis = build_heating_thermostatic_analysis
+      end
+
+      def build_heating_thermostatic_analysis
+        ::Heating::HeatingThermostaticAnalysisService.new(
+          meter_collection: aggregate_school,
+          fuel_type: :storage_heater
+        ).create_model
+      end
+
+      def set_annual_usage_breakdown
+        @annual_usage_breakdown = build_annual_usage_breakdown
+      end
+
+      def build_annual_usage_breakdown
+        ::Usage::AnnualUsageBreakdownService.new(
+          meter_collection: aggregate_school,
+          fuel_type: :storage_heater
+        ).usage_breakdown
+      end
+
+      def analysis_dates
```

Comment:
Do you need to override this method? The base class version looks like it creates the same ranges of dates. It relies on `analysis_start_date` and `analysis_end_date` methods returning the initial dates. Those fetch the right aggregate meter based on value of `advice_page_fuel_type`. So assuming the fuel type is same (there may be `storage_heater`/`storage_heaters` issue) you might not need this. Or may only need to override the `analysis_*_date` methods?